Review

AmedeoSapio opened the pull request:
Basic READMEs for all the components and initial client library.

AmedeoSapio committed:
First README draft

OasisArtisan committed:
Add client library

OasisArtisan committed:
Add other files and submodules

OasisArtisan committed:
Update .gitmodules

OasisArtisan committed:
Remove empty ppp's and add more README files

OasisArtisan committed:
Merge branch 'main' of github.com:AmedeoSapio/p4app-switchML into main

AmedeoSapio committed:
Few minor changes to the READMEs

AmedeoSapio committed:
P4 and Controller READMEs

jklr reviewed (changes requested):
Many folks can be confused by 'switch p4' naming. Let's use distinctive names specific to switchML.

AmedeoSapio committed:
Renamed P4 and controller folders

AmedeoSapio committed:
Mention the Compiler User Guide document

AmedeoSapio committed:
SDE compilation commands

jklr reviewed (approved):
Thanks for addressing the comments. I approve it. We may improve the structure further based on community feedback.

AmedeoSapio committed:
SwitchML P4 code

The P4 code supports both UDP and RoCEv2, with 256B and 1024B
packets using recirculation.
